provider-blackduck/src/main/java/com/synopsys/integration/alert/provider/blackduck/web/PolicyNotificationFilterCustomFunctionAction.java

```diff
@@ -102,15 +102,16 @@ private boolean isJobFilterableByPolicy(Collection<String> notificationTypes) {
     }
 
     private BlackDuckPageResponse<PolicyRuleView> retrievePolicyRules(BlackDuckServicesFactory blackDuckServicesFactory, int pageNumber, int pageSize, String searchTerm) throws IntegrationException {
+        ApiDiscovery apiDiscovery = blackDuckServicesFactory.getApiDiscovery();
```

Comment:
This is an interesting class, it looks like it is used to create full URL's for different data in BD. Originally, we would just reference these URL's using the provided static fields and then merge them with the BD URL, you can see on the left where we do exactly that using blackDuckApiClient.getUrl. Why do we need to do this at all? Can the client not perform this ApiDiscovery task itself? Seems like an unnecessary step users will have to do.